### ISSUE TITLE
Unable to access source from inside nested script

### ISSUE THREAD

**Issue by himanshubhandoh** [elastic/elasticsearch#20728]
**Elasticsearch version**: 4.2.1

**JVM version**:1.8

**Description of the problem including expected versus actual behavior**:
I don't get any results for queries over nested fields, even though the script does work as expected for non nested fields. The reason, as I observed while debugging my wildcard script is that I fetch source() in the script which tends to be empty when I keep my script inside nested, but returns one document per successive call when querying over non nested fields.

My guess is that it has to do with nested fields having their own separate document, and the docId provided for the document when used to fetch source can not be fetched from inside nested field?

**Steps to reproduce**:
The mapping looks like this:
```
properties =  {  
         address = {  
            type=nested,
            properties=            {  
               street=               {  
                  type=string,
                  fielddata=                  {  
                     format=disabled
                  }
               },
               zip=               {  
                  type=long
               }
            }
         },
         age=         {  
            type=long
         },
         designation=         {  
            type=string,
            fielddata=            {  
               format=disabled
            }
         },
         eventTimestamp=         {  
            type=date,
            format=strict_date_optional_time||epoch_millis
         },
         pickupTimestamp=         {  
            type=date,
            format=strict_date_optional_time||epoch_millis
         },
         user=         {  
            type=string,
            index=not_analyzed
         }
      }
   }
```

**The query looks like this:**
```
{
  "size" : 10000,
  "timeout" : 30000,
  "query" : {
    "bool" : {
      "must" : [ {
        "match_all" : { }
      }, {
        "nested" : {
          "query" : {
            "script" : {
              "script" : {
                "inline" : "MyNativePluginScript",
                "lang" : "native",
                "params" : {
                  "params" : "eyJmaWVsZF9uYW1lIjoiYWRkcmVzcy5zdHJlZXQiLCJxdWVyeV9zdHJpbmciOiIqUyoifQ=="
                }
              }
            }
          },
          "path" : "address"
        }
      } ]
    }
  },
  "sort" : [ {
    "eventTimestamp" : {
      "order" : "desc"
    }
  } ]
}
```

**TL;DR**: Script fetches empty source() when kept inside nested construct, but returns valid source() when not nested.

**Comment by jpountz** [comment]
I am torn as to whether we should address it or document this limitation, especially since `_source` access from search scripts is considered a bad practice.

**Comment by himanshubhandoh** [comment]
Is there a workaround? I'm aware it might not be performant, but wonder if it is even theoretically possible currently.

**Comment by martijnvg** [comment]
@himanshubhandoh The `_source` field isn't accessible, because it associated with the root document and not with the nested documents. So you can't access _source from within a `nested` query. What you could do instead is in your script access doc values or stored fields (although doc values is preferred as accessing it is much faster) from within a `nested` query. This should work as doc values / stored fields are being associated with nested documents.

**Comment by himanshubhandoh** [comment]
Well, I wanted to fetch a field that has been analyzed, so doc values is not an option. Is there a way to fetch the raw string from the stored doc (not the index)?

**Comment by martijnvg** [comment]
@himanshubhandoh Maybe stored fields work for you: https://www.elastic.co/guide/en/elasticsearch/reference/2.4/mapping-store.html#mapping-store

**Comment by clintongormley** [comment]
I'm going to close this issue as there is nothing to do here.
